diff -upr SDL.dist/src/joystick/linux/SDL_sysjoystick.c SDL/src/joystick/linux/SDL_sysjoystick.c
--- SDL.dist/src/joystick/linux/SDL_sysjoystick.c	2025-02-02 11:22:38.954904749 +0100
+++ SDL/src/joystick/linux/SDL_sysjoystick.c	2025-02-02 11:23:11.698920801 +0100
@@ -279,6 +279,8 @@ static int IsJoystick(const char *path,
     char *name;
     char product_string[128];
     int class = 0;
+    SDL_joylist_item *item = NULL;
+    int index = 0;
 
     SDL_zero(inpid);
 #ifdef SDL_USE_LIBUDEV
@@ -338,6 +340,29 @@ static int IsJoystick(const char *path,
         SDL_free(name);
         return 0;
     }
+
+    /* X4 hack: duplicate joystick names, append unique index */
+    for (item = SDL_joylist; item != NULL; item = item->next) {
+        size_t name_len = strlen(name),
+               item_len = strlen(item->name);
+
+        if (item_len >= name_len && memcmp(name, item->name, name_len) == 0)
+            index++;
+    }
+
+    if (index > 0)
+    {
+        int len = SDL_strlen(name) + 3;
+        char * new_name = (char *)SDL_malloc(len);
+        if (new_name) {
+            SDL_snprintf(new_name, len, "%s %d", name, index + 1);
+            /* X4 hack: update guid to some unique value */
+            guid->data[15] = index+1;
+            SDL_free(name);
+            name = new_name;
+        }
+    }
+
     *name_return = name;
     *vendor_return = inpid.vendor;
     *product_return = inpid.product;
